openfl/experimental/runtime/local_runtime.py

```diff
@@ -1,232 +1,481 @@
+        """
+        from openfl.experimental.interface import FLSpec
+        aggregator = self._aggregator
+        clones = None
+
+        if self.join_step:
+            clones = [FLSpec._clones[col] for col in self.selected_collaborators]
+            self.join_step = False
+
+        if self.backend == "ray":
+            ray_executor = RayExecutor()
+            ray_executor.ray_call_put(
+                aggregator, flspec_obj,
+                f.__name__, self.execute_agg_steps,
+                clones
+            )
+            flspec_obj = ray_executor.ray_call_get()[0]
+            del ray_executor
+        else:
+            aggregator.execute_func(
+                flspec_obj, f.__name__, self.execute_agg_steps,
+                clones
+            )
+
+        gc.collect()
+        return flspec_obj
+
+    def execute_collab_task(
+        self, flspec_obj, f, parent_func, instance_snapshot, **kwargs
+    ):
+        """
+        Performs
+            1. Filter include/exclude
+            2. Set runtime, collab private attributes , metaflow_interface
+            3. Execution of all collaborator for each task
+            4. Remove collaborator private attributes
+            5. Execute the next function after transition
+
+        Args:
+            flspec_obj  :  Reference to the FLSpec (flow) object
+            f           :  The task to be executed within the flow
+            parent_func : The prior task executed in the flow
+            instance_snapshot : A prior FLSpec state that needs to be restored
+
+        Returns:
+            flspec_obj: updated FLSpec (flow) object
+        """
+
+        from openfl.experimental.interface import (
+            FLSpec,
+        )
+
+        flspec_obj._foreach_methods.append(f.__name__)
+        selected_collaborators = getattr(flspec_obj, kwargs["foreach"])
+        self.selected_collaborators = selected_collaborators
+
+        # filter exclude/include attributes for clone
+        self.filter_exclude_include(flspec_obj, f, selected_collaborators, **kwargs)
+
+        if self.backend == "ray":
+            ray_executor = RayExecutor()
+        # set runtime,collab private attributes and metaflowinterface
+        for col in selected_collaborators:
+            clone = FLSpec._clones[col]
+            # Set new LocalRuntime for clone as it is required
+            # new runtime object will not contain private attributes of
+            # aggregator or other collaborators
+            clone.runtime = LocalRuntime(backend="single_process")
+
+            # write the clone to the object store
+            # ensure clone is getting latest _metaflow_interface
+            clone._metaflow_interface = flspec_obj._metaflow_interface
+
+        for collab_name in selected_collaborators:
+            clone = FLSpec._clones[collab_name]
+            collaborator = self.__collaborators[collab_name]
+
+            if self.backend == "ray":
+                ray_executor.ray_call_put(

```

Comment:
@psfoley As discussed our understanding is that limiting the number of simultaneous processes, in this case, methods of Ray actor scheduled simultaneously, is unlikely to prevent OOM errors

Our experiments demonstrated that Ray Actors are long-lived components and do not release the memory till the Actor goes out of scope.

As per our understanding, attempting to limit the simultaneous processes by 
a) Creating a limited number of Ray Actors and 
b) Deleting the Ray actors once their processing is finished
Will be quite inefficient and counterproductive to the reason why Ray actors were implemented in the first place.